specification/protocol/generate_rest.yaml

```diff
@@ -0,0 +1,89 @@
+openapi: 3.1.0
+info:
+  title: Open Inference API for text generation
+  description: Open Inference API for text generation
+  version: 1.0.0
+components:
+  schemas:
+    GenerateRequest:
+      type: object
+      required: 
+        - text_input
+      properties:
+        text_input:
+          type: string
+        parameters:
+          $ref: '#/components/schemas/GenerateParameters'
+    GenerateParameters:
```

Comment:
I think we might still want to define a list of common properties and then in addition user can still add custom parameters.